### PR TITLE
Lamp value range validation

### PR DESCRIPTION
**Pull request by Lakitna**
#138

### CODE REVIEW

File: pytradfri/device.py [hunk 1]
```diff
@@ -263,7 +263,7 @@ def _value_validate(self, value, rnge, identifier="Given"):
         """
         Make sure a value is within a given range
         """
-        if value < rnge[0] or value > rnge[1]:
+        if value is not None and (value < rnge[0] or value > rnge[1]):
```
Comment (lwis) [inline pytradfri/device.py]: `if value and (value < rnge[0] or value > rnge[1]):`

Comment (lwis) [inline pytradfri/device.py]: Could we not have this as;

`if value is None or (value < rnge[0] or value > rnge[1]):`

Comment (Lakitna) [reply pytradfri/device.py]: That would raise the error if `value == None`. Here we never want `None` to raise an error.

**Comment by lwis** [comment]
Is `None` a valid value for all callers of `_value_validate`?

**Comment by Lakitna** [comment]
It should be, but I'll check to be sure.

**Comment by Lakitna** [comment]
Yes, `None` is valid for every instance. It's simply ignored by the lamp without any callback.

**Comment by ggravlingen** [comment]
Nice!

**Comment by ggravlingen** [comment]
@Lakitna do you need this on pypi for your use purposes?

**Comment by Lakitna** [comment]
Scratch that, I made a mistake. Please revert the merge.

This only works for brightness. For every other value it is changed to `0`.

**Comment by Lakitna** [comment]
This makes things a bit more complicated.

I still think it's a good idea to be able to set only a single value, even in compound values (XY, HSB).

I've got an idea, I'll work on it and make a suggestion via a commit in a bit.

**Pull request merged**